flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/rules/dataset/DataSetAggregateRule.scala

```diff
@@ -44,17 +45,24 @@ class DataSetAggregateRule
     val inputType = agg.getInput.getRowType()
 
     // add grouping fields, position keys in the input, and input type
-    val aggregateFunction = AggregateFactory.createAggregateInstance(agg.getAggCallList,
-        inputType, grouping)
+    val aggregateResult = AggregateUtil.createOperatorFunctionsForAggregates(agg,
```

Comment:
`createOperatorFunctionsForAggregates` only uses `agg.getAggCallList()`. 
Can you change `createOperatorFunctionsForAggregates` to expect `List<AggCall>` instead of `FlinkAggregate`?